torch/fx/symbolic_trace.py

```diff
@@ -307,13 +333,48 @@ def forward(*args, **kwargs):
             # Seems to be a mypy limitation: https://github.com/python/mypy/issues/2427
             torch.nn.Module.__getattr__ = module_getattr_wrapper  # type: ignore
             torch.nn.Module.__call__ = module_call_wrapper
+            # Install _wrap_fn_impl so that we can properly record calls to
+            # leaf functions, i.e. functions decorated with @fx.wrap
+            global _wrap_fn
+            _wrap_fn = _wrap_fn_impl
```

Comment:
I wonder if we could make _wrap_fn_impl a class method. That would have two benefits: 
One is that find_proxy will not be needed anymore (since _wrap_fn_impl bounds to the proxy object). Although, you may still want to keep it to distinguish cases where no proxy is used .
But more importantly, this enables overriding _wrap_fn_impl in tracer subclasses. We already have a usecase that requires overriding _wrap_fn_impl (to enable support for symtrace).